src/compiler/checker.ts

```diff
@@ -10606,6 +10609,29 @@ namespace ts {
             return false;
         }
 
+        function elaborateDidYouMeanToCallOrConstruct(node: Expression, source: Type, target: Type): boolean {
+            const callSignatures = getSignaturesOfType(source, SignatureKind.Call);
+            const constructSignatures = getSignaturesOfType(source, SignatureKind.Construct);
+            for (const signatures of [constructSignatures, callSignatures]) {
+                if (length(signatures)) {

```

Comment:
Do you need the `length()` call with `some`?

Reply:
Probably not.